src/java.base/share/classes/javax/crypto/JceSecurityManager.java

```diff
@@ -100,11 +102,13 @@ CryptoPermission getCryptoPermission(String alg) {
         // javax.crypto.* packages.
         // NOTE: javax.crypto.* package maybe subject to package
         // insertion, so need to check its classloader as well.
-        Class<?>[] context = getClassContext();
+        List<StackFrame> stack = StackWalker.getInstance(
+                StackWalker.Option.RETAIN_CLASS_REFERENCE)
+                .walk((s) -> s.collect(Collectors.toList()));
```

Comment:
StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE) will require a permission check.
As long as the SecurityManager is still functional, doesn't this mean that creating the StackWalker should be performed in a doPrivileged? If so maybe it should be done in a (possibly static) initializer. Or is it intentional to check that the caller (and the whole calling stack) posses the `RuntimePermission("getStackWalkerWithClassReference")`?

Reply:
Good catch, thanks.